sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1041,8 +1041,8 @@ _pi_queue::resetCommandList(pi_command_list_ptr_t CommandList,
       ze_result_t ZeResult =
           ZE_CALL_NOCHECK(zeEventQueryStatus, ((*it)->ZeEvent));
       if (ZeResult == ZE_RESULT_SUCCESS) {
-        std::move(it, it, std::back_inserter(EventListToCleanup));
-        EventList.erase(it, it);
+        EventListToCleanup.push_back(std::move((*it)));
+        it = EventList.erase(it, it);
```

Comment:
What if it was the last event and `it` now points to `end` iterator? Wouldn't `it++` of it fail?

Reply:
Even a removal at the end of the list works as expected.

Reply:
My question wasn't about the removal but about the `it++` executed along the loop backedge after said removal is done. Is that ok as well?

Reply:
It was not OK to simply update the iterator.
Further digging revealed why in an earlier version of the code some events in the incoming list were skipped during examining for being reset. Now, the for-loop does not increment the iterator. If an event is removed the iterator is set to the element after the one being removed, and if the event is not erased then the iterator is incremented. This avoid the double ++ that was occurring before, whenever an event was removed.